src/test/java/core/basesyntax/dao/ProductDaoImplTest.java

```diff
@@ -0,0 +1,81 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Product;
+import java.util.List;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ProductDaoImplTest {
+    private static ProductDao productDao;
+
+    @BeforeClass
+    public static void setProductDao() {
+        productDao = new ProductDaoImpl();
+    }
+
+    @After
+    public void afterEachTest() {
+        Storage.products.clear();
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void addNullProduct_NotOk() {
```

Comment:
Fix **all** methods in **all** classes
![image](https://user-images.githubusercontent.com/20326858/229760213-47338380-872e-4539-a6f7-75a575ffc247.png)